app/mutations/address_pools/release.rb

```diff
@@ -15,7 +15,7 @@ def validate
     end
 
     def execute
-      @pool.delete!
+      @pool.release!
```

Comment:
Fixed to use `@pool.cleanup`, which will `delete!` the AddressPool if `orphaned?`